### PR TITLE
[NV-1803] Fix: Update `env.sh` to Properly Read Runtime Vars

### LINKED ISSUE #2939: 🐛 Bug Report: `API_ROOT_URL` not being read from .env file
### 📜 Description

Version - 0.12.0

The frontend serving app ie. FRONT_BASE_URL makes calls to API_ROOT_URL. 

Issue is, `API_ROOT_URL` variable value doesn't seem to be read from `.env` file, but rather it seems to be hardcoded to localhost.

### 👟 Reproduction steps

1. Fetch the file https://github.com/novuhq/novu/blob/next/docker/local/deployment/docker-compose.yml
2. Edit compose file and replace versions 0.11.0 with 0.12.0 (since I wanted to make an upgrade to use new featues)
3. In the same directory, have an `.env` file.
4. for variable `API_ROOT_URL`, try using a different hostname -- other than localhost
5. build image & start containers `docker-compose -f docker-compose.yml up`
6. Open `FRONT_BASE_URL` in a browser
7. The frontend serving app (FRONT_BASE_URL) should use localhost for `API_ROOT_URL`. <-- that's the bug.

### 👍 Expected behavior

Expected behavior: API_ROOT_URL value to be read from `.env` file

### 👎 Actual Behavior with Screenshots

Actual behavior: API_ROOT_URL is being set as localhost, although the value in `.env` is set to my public URL of EC2.
image

### 📃 Provide any additional context for the Bug.

Original discord discussion: https://discord.com/channels/895029566685462578/1081138836605775912/1081183315249336360

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Contributing Guidelines?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by JoeyEamigh**
### What change does this PR introduce?

The `env.sh` script should loop through the `.env` file and check if there are runtime variables present in the current shell. If there are, it should write them to the `env-config.js` file.

The issue was that instead of continuing the loop if there wasn't a value in the shell, the loop would continue if there wasn't a value in `.env`, requiring the environment variables to be set at build time and defeating the whole purpose of the script.

Now it will continue only if there is no value in the shell, and no default value in `.env`.

### Why was this change needed?

Closes #2939.

### Other information

I also added other env vars that were used in `index.ts` but were not present in the `.env` file for it to work with this `env.sh` script.

### CODE REVIEW

File: apps/web/env.sh [hunk 1]
```diff
@@ -5,34 +5,34 @@ rm -rf ./env-config.js
 touch ./env-config.js
 
 # Add assignment
-echo "window._env_ = {" >> ./env-config.js
+echo "window._env_ = {" >>./env-config.js
```
Comment (LetItRock) [inline apps/web/env.sh:8]: We should do exactly the same also for the `widget` app. Could you adjust that file too?

**Comment by vipulsharma144** [comment]
Hey! Can you folks suggest any alternatives till the time this is merged? I am unable to use the web app using docker deployment due to this bug. It always points to localhost:3000.

**Comment by JoeyEamigh** [comment]
The widget `env.sh` seemed to be different and actually working, however I did update it to be the same.

**Comment by JoeyEamigh** [comment]
For @vipulsharma144 and others who would like a working build until this gets merged, I have `ghcr.io/joeyeamigh/novu/web:nv-1803` public so anyone can pull it.

**Review by scopsy — APPROVED** [review]

**Comment by katyaterletskaya** [comment]
Hi @scopsy 😊 Do you have any plans to publish a new image containing this fix?

**Comment by vishnu-kyatannawar** [comment]
> Hi @katyaterletskaya the new version will be available on mid-next week pray

@scopsy any update on new version?

**Comment by katyaterletskaya** [comment]
Hi @scopsy 😊 Would it be possible to add a `0.13.0` tag to the images?

**Comment by scopsy** [comment]
@katyaterletskaya yes! Was just about to publish it :P

**Pull request merged**